nixos/modules/services/computing/boinc/client.nix

```diff
@@ -6,6 +6,24 @@ let
   cfg = config.services.boinc;
   allowRemoteGuiRpcFlag = optionalString cfg.allowRemoteGuiRpc "--allow_remote_gui_rpc";
 
+  # A package providing the boinc_client that will actually be executed.
+  execPkg = if cfg.useFHSEnv then
+    pkgs.buildFHSUserEnv {
+      name = "boinc_client";
+      targetPkgs = p: [ cfg.package ] ++ optional cfg.virtualbox.enable cfg.virtualbox.package;
+      multiPkgs = p: optional cfg.gpu.nvidia.enable cfg.gpu.nvidia.package;
+      runScript = "/bin/boinc";
+      extraBuildCommands = ''
+        if [ -e lib/vendors ]; then
+          chmod +w etc
+          mkdir -p etc/OpenCL
+          ln -s $(realpath lib/vendors) etc/OpenCL/vendors
+          chmod -w etc
+        fi
+      '';
```

Comment:
I'm not sure on OpenCL story but we should probably see how/if it works on NixOS currently, what paths are used etc. Probably we should have `nvidia-x11` installing it in the correct place.

BTW, you can make this less hacky by adding a bogus package with symlink to the environment:

```
pkg = runCommand "nvidia-opencl" {} ''
  mkdir -p $out/etc/OpenCL
  ln -s ${nvidia-x11}/lib/vendors $out/etc/OpenCL
'';